lib/extractor/default.nix

```diff
@@ -106,15 +116,16 @@ let
       # only use selected interpreters
       in optionalString (pyVersions == [] || elem v pyVersions) ''
         echo "extracting metadata for python${v}"
-        out_file=$out/python${v}.json ${py}/bin/python -c "${setuptools_shim}" install &> $out/python${v}.log || true
+        SETUPTOOLS_USE_DISTUTILS=stdlib out_file=$out/python${v}.json ${py}/bin/python -c "${setuptools_shim}" install &> $out/python${v}.log || true
       ''
     ))}
   '';
   script_single = py: ''
     chmod +x setup.py || true
     mkdir $out
     echo "extracting dependencies"
-    out_file=$out/python.json ${py}/bin/python -c "${setuptools_shim}" install &> $out/python.log || true
+    # Restore old setuptools behaviour (since # https://github.com/pypa/setuptools/commit/b6fcbbd00cb6d5607c9272dec452a50457bdb292), # to keep it working with mach-nix.
```

Comment:
I don't think the hashes (`#`) inside the comment belongs there? Was the line perhaps wrapped and then joined without removing the comment chars? (I prefer wrapping the long line.)

Reply:
you're right.
I've also pulled the comment out of the shell script - no need to store it in every nix log.